src/modules/handlers/ManifestHandler.js

```diff
@@ -19,40 +19,88 @@ DB.version(1).stores({
 });
 
 function SetNextManifestCheck() { localStorage.setItem('nextManifestCheck', new Date().getTime() + (1000 * 60 * 60)); }
-export function StoreManifest(callback) {
-  callback({ status: 'storingManifest', statusText: 'Storing Manifest...', error: false, loading: true, manifestMounted: false });
-  Promise.all([
-    DB.table('DestinyActivityDefinition').toCollection().first(),
-    DB.table('DestinyActivityTypeDefinition').toCollection().first(),
-    DB.table('DestinyActivityModeDefinition').toCollection().first(),
-    DB.table('DestinyCollectibleDefinition').toCollection().first(),
-    DB.table('DestinyPresentationNodeDefinition').toCollection().first(),
-    DB.table('DestinyRecordDefinition').toCollection().first(),
-    DB.table('DestinyInventoryItemLiteDefinition').toCollection().first(),
-    DB.table('DestinyObjectiveDefinition').toCollection().first(),
-    DB.table('DestinyProgressionDefinition').toCollection().first(),
-    DB.table('DestinyTalentGridDefinition').toCollection().first(),
-    DB.table('DestinyVendorDefinition').toCollection().first()
-  ]).then(async (values) => {
+
+const manifestLoadAttempts = 0;
+const maxLoadAttempts = 3;
+export async function StoreManifest(callback) {
+  try {
+    if (manifestLoadAttempts > maxLoadAttempts) {
+      throw `Manifest is corrupted after ${maxLoadAttempts} attempts.`;
+    }
+    manifestLoadAttempts++;
```

Comment:
This doesn't work due to it trying to change the const value of `manifestLoadAttempts`, easy fix is to change this to a let.